x/blocksdk/keeper/msg_server.go

```diff
@@ -0,0 +1,47 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/skip-mev/block-sdk/x/blocksdk/types"
+)
+
+var _ types.MsgServer = MsgServer{}
+
+// MsgServer is the wrapper for the x/blocksdk module's msg service.
+type MsgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the x/blocksdk MsgServer interface.
+func NewMsgServerImpl(keeper Keeper) *MsgServer {
+	return &MsgServer{Keeper: keeper}
+}
+
+// UpdateLane implements the message service for updating a lane that exists in the store.
+func (m MsgServer) UpdateLane(goCtx context.Context, msg *types.MsgUpdateLane) (*types.MsgUpdateLaneResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// ensure that the message signer is the authority
+	if msg.Authority != m.Keeper.GetAuthority() {
+		return nil, fmt.Errorf("this message can only be executed by the authority; expected %s, got %s", m.Keeper.GetAuthority(), msg.Authority)
+	}
+
+	oldLane, err := m.GetLane(ctx, msg.Lane.Id)
+	if err != nil {
+		return nil, fmt.Errorf("lane with ID %s not found", msg.Lane.Id)
+	}
+
+	if oldLane.Order != msg.Lane.Order {
```

Comment:
Why are we asserting this? This makes sense to me, but I forgot exactly where we made this decision?

Reply:
UpdateLane only lets you update a single late, so if you modify the Order, you'll inevitably be taking an order that is already used by another lane or make the order sequence be non-monotonically increasing